app/connector/kafka/src/main/java/io/syndesis/connector/kafka/KafkaMetaDataRetrieval.java

```diff
@@ -71,46 +87,64 @@ protected SyndesisMetadata adapt(CamelContext context, String componentId, Strin
-                                                              .withName("kafkas.kafka.strimzi.io")
-                                                              .withPlural("kafkas")
-                                                              .withScope("Namespaced")
-                                                              .withVersion("v1beta1")
-                                                              .build();
-
-            final List<Map<String, ?>> items = (List<Map<String, ?>>) client.customResource(build).list().get("items");
-
-            for (Map<String, ?> item : items) {
-                //Extract an identifier of this broker
-                final Map<?, ?> metadata = (Map<?, ?>) item.get("metadata");
-                String id = metadata.get("namespace") + "::" + metadata.get("name");
-
-                //Add the list of addresses for this item
-                Map<?, ?> status = (Map<?, ?>) item.get("status");
-                List<Map<String,
-                            List<Map<String, Object>>>> listeners =
-                    (List<Map<String, List<Map<String, Object>>>>) status.get("listeners");
-                listeners.stream().filter(typesAllowed).forEach(
-                    listener -> getAddress(listener, brokers, id));
-            }
+            client.customResourceDefinitions().list().getItems()
+                .stream().filter(isKafkaBroker)
+                .forEach(kafka -> processKafkaCRD(brokers, client, kafka));
+        } catch (Throwable t) {
+            LOG.warn("Couldn't auto discover any broker.", t);
         }
 
         Map<String, List<PropertyPair>> dynamicProperties = new HashMap<>();
         dynamicProperties.put("brokers", brokers);
         return new SyndesisMetadataProperties(dynamicProperties);
     }
 
+    /**
+     * For each Kafka container definition found, process it.
+     *
+     * @param brokers
+     * @param client
+     * @param crd
+     */
+    private void processKafkaCRD(List<PropertyPair> brokers, KubernetesClient client, CustomResourceDefinition crd) {
+        KafkaResourceList list = client.customResources(crd,
+            KafkaResource.class,
+            KafkaResourceList.class,
+            KafkaResourceDoneable.class).inAnyNamespace().list();
+
+        for (KafkaResource item : list.getItems()) {
+            processKafkaResource(brokers, item);
+        }
+    }
+
+    /**
+     * For each Kafka resource found on Kubernetes, extract the listeners and
+     * add them to the brokers list.
+     *
+     * @param brokers
+     * @param item
+     */
+    @SuppressWarnings("unchecked")
+    private void processKafkaResource(List<PropertyPair> brokers, KafkaResource item) {
+        //Extract an identifier of this broker
+        final ObjectMeta metadata = item.getMetadata();
+        String id = metadata.getNamespace() + "::" + metadata.getName();
+
+        List<Map<String, List<Map<String, Object>>>> listeners =
```

Comment:
Try to make warning suppressions as precise as they can be.

```suggestion
        @SuppressWarnings("unchecked")
        List<Map<String, List<Map<String, Object>>>> listeners =
```

Reply:
ok